jax/_src/interpreters/batching.py

```diff
@@ -649,7 +649,7 @@ def untile_axis(out, axis: int | None):
 
   @lu.transformation2
   def _map_to_tile(f, *args_flat):
-    sizes = (x.shape[i] for x, i in safe_zip(args_flat, in_axes_flat) if i is not None)
+    sizes = (x.shape[i] for x, i in unsafe_zip(args_flat, in_axes_flat, strict=True) if i is not None)
```

Comment:
I would argue that this is harder to reason about than having `safe_zip()` :)

unsafe_zip(..., strict=True) is a double negative!

Reply:
Right – that's a good reason why we should get rid of `unsafe_zip` here and just call it `zip` (but Matt disagrees).

Reply:
I agree with Matt. `safe_zip` is good to have. It's very clear about it's intention and when you see it, you know exactly what the code is doing.